pact/dao/dao.pact

```diff
@@ -0,0 +1,279 @@
+(enforce-pact-version "3.7")
+
+(namespace (read-msg 'dao-ns))
+
+(module utils-v1 GOVERNANCE
+  (defcap GOVERNANCE () false)
+  (defun now:time () (at "block-time" (chain-data)))
+
+  (defun today:time ()
+    (parse-time "%Y-%m-%d" (format-time "%Y-%m-%d" (now))))
+
+  (defun yesterday:time () (add-time (today) (days (- 1))))
+
+  (defun tomorrow:time () (add-time (today) (days 1)))
+
+  (defun btwn-incl:bool (start end ts)
+    (enforce (>= end start) (format "bounding start {} must come before end {}" [start end]))
+    (and (>= ts start) (<= ts end)))
+  (defun btwn-excl:bool (start end ts)
+    (enforce (>= end start) "bounding end comes before start")
+    (and (> ts start) (< ts end)))
+
+  (defun this-block:integer () (at 'block-height (chain-data)))
+
+  (defconst BLOCKHEIGHT_PER_MINUTE 2)
+  (defconst BLOCKHEIGHT_PER_HOUR 120)
+  (defconst BLOCKHEIGHT_PER_DAY 2880)
+
+  (defun row-with-key (tbl k)
+    (let ((r (read tbl k)))
+      [k r]))
+
+  )
+
+(module dao-v1 GOVERNANCE
+  @doc
+  "This is the start of the KDA DAO, not the end. \
+  \It's not even something that should set a precedence. \
+  \It intentionally doesn't incentivize participation. \
+  \It doesn't even have a function to de-stake. \
+  \Instead a community approved(ish) contract upgrade is needed to add functionality. \
+  \The goal is to get KDA's guardians and ambassadors together, and make something better. \
+  \Guardians are the stakers, with governance rights to approve and apply upgrades. \
+  \Proposed upgrades have a 1 day cooldown before they can be voted on and applied. \
+  \Proposals timeout after 3 days, and need to be re-submitted thereafter. \
+  \During this period, ambassadors can exercise their oversite: \
+  \a majority vote by the ambassadors can freeze the DAO for 7 days."
+
+  (use utils-v1)
+  (defconst DAO_MODULE_NAME "dao-v1")
+  (defconst DAO_ACCT_NAME "dao-v1") ; we'll change this
+  (defun DAO_ACCT_BALANCE () (coin.get-balance DAO_ACCT_NAME))
+  (defcap INTERNAL ()
+    "mark some functions as internal only"
+    true)
+
+  ; Approved upgrades need to be at least this old to occur.
+  ; This gives the ambassadors time to respond
```

Comment:
Consider making these comments docs. format is `(defconst NAME VALUE DOC)`, so e.g. 
```
  (defconst APPROVAL_COOLDOWN (days 1)
    "Approved upgrades need to be at least this old to occur.")
```